src/Tasks/Microsoft.NET.Build.Tasks/ResolveTargetingPackAssets.cs

```diff
@@ -208,13 +267,43 @@ private void AddNetStandardTargetingPackAssets(ITaskItem targetingPack, string t
             string targetingPackDllFolder,
             ITaskItem targetingPack, List<TaskItem> referenceItems, List<TaskItem> analyzerItems)
         {
-            XDocument frameworkListDoc = XDocument.Load(frameworkListPath);
-
             string profile = targetingPack.GetMetadata("Profile");
 
+            StringBuilder hashKey = new(nameof(ResolveTargetingPackAssets) + nameof(AddItemsFromFrameworkList));
```

Comment:
Same feedback here about making sure we're caching all the inputs and adding test coverage for that if possible.

Reply:
This one is harder since it's internal but I'll see what I can do.

Reply:
It was indeed harder but a bit of refactoring made it work ok. Look good now?